docs/LabHome/Onboarding/coding-guide.md

```diff
@@ -0,0 +1,91 @@
+---
+layout: default
+title: Best practices for coding for scientific computing
+parent: Lab Basics
+nav_order: 3
+has_toc: true
+---
+
+# Best practices for coding for scientific computing at PennLINC
+
+This is a guide for coding for scientific computing at PennLINC. People may have different styles
+in coding; however, there are a few guidelines we recommend here in order to enhance code's reproducibility, reusability, to make it less prone to mistakes, etc. In other words, your 
+code achieves exactly what you wanted; after several months, when looking at your code, you 
+can still quickly get what you coded; you or other lab members can also reuse your code for other projects in the future.
+
+We will describe the guidelines in three stages: 
+
+1. [Before you start coding for a project](#before-you-start-coding-for-a-project)
+2. [During coding](#coding-itself)
+3. [After you code a bit](#after-you-coded-a-bit)
+
+We end here with some [final words](#final-words).
+
+## Before you start coding for a project
+
+* Initialize a GitHub repository for your project [under PennLINC GitHub group](https://github.com/PennLINC). Use Git and GitHub from the beginning of your project, instead of not using them until the end stage of the project. You'll appreciate the code version control when you hope to see the changes of your code. If you are a bit hesitated for making your code fully public at the beginning of the project, that's okay - you can always choose to make the GitHub repository "Private".
```

Comment:
```suggestion
* Create a GitHub repository for your project under the [PennLINC GitHub group](https://github.com/PennLINC). Use Git and GitHub from the beginning of your project, instead of waiting until the end stage of the project. You'll appreciate the version control when you need to look back at changes you made to your code throughout the project's lifespan. If you are worried about making your code fully public at the beginning of the project, that's okay - you can always choose to make the GitHub repository "Private".
```